corehq/pillows/case_search.py

```diff
@@ -88,7 +88,7 @@ def _get_case_properties(doc_dict):
     ]
     dynamic_case_properties = dict(doc_dict['case_json'])
 
-    dynamic_mapping = [{'key': key, VALUE: value} for key, value in dynamic_case_properties.items()]
+    dynamic_mapping = [{'key': key, VALUE: str(value)} for key, value in dynamic_case_properties.items()]
```

Comment:
The only thing I'd ask is do we want think to leave a record of it somewhere (log, sentry,  etc) if we think in the future we'll make it so this can't happen?